Lib/test/test_os.py

```diff
@@ -4534,6 +4534,31 @@ def test_fork(self):
         assert_python_ok("-c", code)
         assert_python_ok("-c", code, PYTHONMALLOC="malloc_debug")
 
+    @unittest.skipUnless(sys.platform in ("linux", "darwin"),
+                         "Only Linux and macOS detect this today.")
+    def test_fork_warns_when_non_python_thread_exists(self):
+        code = """if 1:
+            import os, threading, warnings
+            from _testcapi import _spawn_pthread_waiter, _end_spawned_pthread
+            _spawn_pthread_waiter()
+            try:
+                with warnings.catch_warnings(record=True) as ws:
+                    warnings.filterwarnings(
+                            "always", category=DeprecationWarning)
+                    if os.fork() == 0:
+                        assert len(ws) == 0  # Warn only the parent
```

Comment:
I cleaned it up a bit anyways. assert_python_ok doesn't appear to reconstruct -O flags on the command line, but at least explicitly having it set PYTHONOPTIMIZE=0 means the asserts should always work.

there was nothing checking for a child process error anyways, it could've gotten lost. fixed now.